msal/src/main/java/com/microsoft/identity/client/AuthenticationActivity.java

```diff
@@ -130,22 +131,38 @@ private void warmUpCustomTabs() {
         mCustomTabsIntent.intent.setPackage(mChromePackageWithCustomTabSupport);
     }
 
-    @NonNull
-    private CustomTabsServiceConnection createCustomTabsServiceConnection() {
-        return new CustomTabsServiceConnection() {
-            @Override
-            public void onCustomTabsServiceConnected(ComponentName name, CustomTabsClient client) {
-                mCustomTabsServiceIsBound = true;
-                mCustomTabsClient = client;
-                mCustomTabsClient.warmup(0L);
-                mCustomTabsSession = mCustomTabsClient.newSession(null);
+    private static class MsalCustomTabsServiceConnection extends CustomTabsServiceConnection {
```

Comment:
I'm a little bit worried about the implementation 1) connection class having reference to enclosing class doesn't look good :) 2) in line 121, we call bindCustomTabsService, which is an asynchronous call, but we pass in the customtab session right after it to intent builder, is it possible that it's not connected yet? 3) I don't think we're using warm up at all. If you look at the onResume, it's creating a new customTabIntent and that is the intent used to launch URL.